Sofa/framework/Config/CMakeLists.txt

```
@@ -130,6 +130,20 @@ set(SOFACONFIG_LINK_OPTIONS "")
 set(SOFACONFIG_LINK_OPTIONS_DEBUG "")
 set(SOFACONFIG_LINK_OPTIONS_RELEASE "")
 
+include(CheckIPOSupported)
+check_ipo_supported(RESULT IS_IPO_SUPPORTED OUTPUT IPO_ERROR)
+
+## Link-time optimization
+if (IS_IPO_SUPPORTED)
+    # Focus on max speed with link-time optimization
+    option(SOFA_ENABLE_LINK_TIME_OPTIMIZATION "Enable interprocedural optimization" OFF)
+    if (SOFA_ENABLE_LINK_TIME_OPTIMIZATION)
+        set(CMAKE_INTERPROCEDURAL_OPTIMIZATION TRUE)
```

```suggestion
        set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_RELEASE TRUE PARENT_SCOPE)
```
OK I think it should do the trick
`CMAKE_INTERPROCEDURAL_OPTIMIZATION` value will only be set for the present scope (aka Sofa.Config) ; if we set the additional `PARENT_SCOPE` argument , this option will propagate to the other scopes.
The other method would be to move this option directly to the main CMakeFile.txt at the root of the project.

I just tried on gcc and the flags `flto=auto -fno-fat-lto-objects` are added sucessfully
And I added `_RELEASE` suffix to only enable the LTO for Release mode

---

It makes sense. Thanks.
I'll try on Windows, and check if the flags are still required.

---

I confirm your suggestion adds the /GL and /LTCG flags.
However, it is available only in release, not in RelWithDebInfo. Would that make sense to add it also for RelWithDebInfo?

---

In my mind, one should use LTO only for "production-ready" build so IMO it makes sense to use it only for Release. (I see RelWithDebInfo as a speedy debug build 😅)
But I am not against setting it for RelWithDebInfo as well, especially because IPO is only an option.